bin/nova-hypervisor-usage

```diff
@@ -192,18 +196,35 @@ def compute_capability(flavor, hypervisors):
     max_count_wo_disabled = 0
 
     for node in hypervisors:
-        vms_by_cpus = int((node.vcpus_max - node.vcpus_used) / flavor.vcpus)
         vms_by_mem = node.memory_mb_free / flavor.ram
-        n_vms = min(
-            vms_by_cpus,
-            vms_by_mem)
-        if n_vms < 0: n_vms = 0
-
-        max_vms_by_cpus = int((node.vcpus_max) / flavor.vcpus)
         max_vms_by_mem = node.memory_mb / flavor.ram
-        n_maxvms = min(
-            max_vms_by_cpus,
-            max_vms_by_mem)
+
+        if 'gpu' in flavor.name:
+            if 'gpuT4' in flavor.name:
+                n_maxvms = min(
+                    DEFAULT['GPU_T4_SLOTS'],
+                    max_vms_by_mem
+                    )
+            elif 'gpuP4' in flavor.name:
+                n_maxvms = min(
+                    DEFAULT['GPU_P4_SLOTS'],
+                    max_vms_by_mem
+                    )
+            #else throw exception
```

Comment:
```suggestion
```

Reply:
I was wondering if was worthed to be a bit more defensive, and manage the error when there is a flavor named *gpu* but neither T4 or P4...